### ISSUE TITLE
APIs for other apps

### ISSUE THREAD

**Issue by georgehrke** [nextcloud/calendar#812]
The calendar app should provide APIs to allow other apps to extend the calendar's functionality.
Things that should be extendable:
- [ ] custom views
- [ ] advanced events editor

anything else?

**Comment by Peterede** [comment]
We are looking at extending the calendar UI ourselves and are happy to help implement this.  Do you have a preferred hooking framework for other apps to modify the calendar?

**Comment by georgehrke** [comment]
@Peterede One of the biggest issues that's keeping us from implementing this is the fact that we are still running on Angular 1, which is not being developed any further and is soon to be end of live.
We are looking forward to move to Vue: #540 

Once we moved to Vue, we will most likely provide a js API to hook in your own vue modules and a hock in our view controller to allow you to load your own javascript.

**Comment by ImaCrea** [comment]
hello there, hope all is fine. I was wondering about the status of this issue please :)
we'd love to have this kind of features https://github.com/nextcloud/calendar/issues/2136 implemented after 🤗

all the good and happy new year

**Comment by towo** [comment]
Hey @ChristophWurst, is this on the roadmap? Would incentives help? ;)

**Comment by ChristophWurst** [comment]
https://docs.nextcloud.com/server/stable/developer_manual/digging_deeper/groupware/calendar.html

**Comment by tcitworld** [comment]
The issue is more about extending the front-end views though. That link is not really relevant…yet.

**Comment by towo** [comment]
Exactly, I was essentially riding on the back of #4948, as that's something we'd very much like to deploy.

**Comment by luflow** [comment]
I am also currently looking for an easy way in the frontend to add a new entry to this context menu of an event to link to my own nextcloud app:

<img width="778" height="366" alt="Image" src="https://github.com/user-attachments/assets/a12d22ff-67ab-47ca-955c-c2b56549abcc" />
